packages/squiggle-lang/src/js/index.ts

```diff
@@ -75,147 +60,234 @@ export function resultMap<a, b, c>(
   }
 
-  sample(): result<number, error> {
+  sample(): result<number, distributionError> {
     return Constructors_sample({ env: this.env }, this.t);
   }
 
-  pdf(n: number): result<number, error> {
+  pdf(n: number): result<number, distributionError> {
     return Constructors_pdf({ env: this.env }, this.t, n);
   }
 
-  cdf(n: number): result<number, error> {
+  cdf(n: number): result<number, distributionError> {
     return Constructors_cdf({ env: this.env }, this.t, n);
   }
 
-  inv(n: number): result<number, error> {
+  inv(n: number): result<number, distributionError> {
     return Constructors_inv({ env: this.env }, this.t, n);
   }
 
-  normalize(): result<Distribution, error> {
+  normalize(): result<Distribution, distributionError> {
     return this.mapResultDist(
       Constructors_normalize({ env: this.env }, this.t)
     );
   }
 
-  toPointSet(): result<Distribution, error> {
+  shape() : result<shape, distributionError> {
+    let pointSet = toPointSet(this.t, {xyPointLength: this.env.xyPointLength, sampleCount: this.env.sampleCount}, null);
+    if(pointSet.tag === "Ok"){
+      let distribution = pointSet.value;
+      if(distribution.tag === "Continuous"){
+        return Ok({
+          continuous: shapePoints(distribution.value),
+          discrete: []
+        })
+      }
+      else if(distribution.tag === "Discrete"){
+        return Ok({
+          discrete: shapePoints(distribution.value),
+          continuous: []
+        })
+      }
+      else if(distribution.tag === "Mixed"){
+        return Ok({
+          discrete: shapePoints(distribution.value.discrete),
+          continuous: shapePoints(distribution.value.continuous)
+        })
+      }
+    }
+    else {
+      return pointSet
+    }
+  }
+
+  toPointSet(): result<Distribution, distributionError> {
     return this.mapResultDist(
       Constructors_toPointSet({ env: this.env }, this.t)
     );
   }
 
-  toSampleSet(n: number): result<Distribution, error> {
+  toSampleSet(n: number): result<Distribution, distributionError> {
     return this.mapResultDist(
       Constructors_toSampleSet({ env: this.env }, this.t, n)
     );
   }
 
-  truncate(left: number, right: number): result<Distribution, error> {
+  truncate(left: number, right: number): result<Distribution, distributionError> {
     return this.mapResultDist(
       Constructors_truncate({ env: this.env }, this.t, left, right)
     );
   }
 
-  inspect(): result<Distribution, error> {
+  inspect(): result<Distribution, distributionError> {
     return this.mapResultDist(Constructors_inspect({ env: this.env }, this.t));
   }
 
-  toString(): result<string, error> {
-    return Constructors_toString({ env: this.env }, this.t);
+  toString(): string {
+    let result = Constructors_toString({ env: this.env }, this.t);
+    if(result.tag  === "Ok"){
+      return result.value
+    }
+    else {
+      return distributionErrorToString(result.value)
+    }
```

Comment:
Can we wrap this in a function called ``mapResultString``, similar to what we do for the other functions?

Reply:
Do I need to? mapResultDist returns a distribution, this only returns a string? What would `mapResultString` do? `mapResultDist` wraps a `genericDist` in a `Distribution`. So I'm confused what you're expecting `mapResultString` to do? Especially considering there isn't a result here?

Reply:
sorry; you're right about not mapping string, but, we could still have something to do the error mapping in one line.

Like,
``mapError(distributionErrorToString)``

Reply:
I feel like it's not quite a "map" because it never returns a result. I think in this particular case it's more doing what would be haskell's [`fromRight`](https://hackage.haskell.org/package/base-4.16.1.0/docs/Data-Either.html#v:fromRight). I have a feeling it might be good to leave this as is until I see a second `fromRight` like pattern in this file, then I might refactor

Reply:
In the ``squiggle-lang`` codebase, we have a function for E.R2.errMap. We don't use the terminology ``Right`` in Rescript, so ``fromRight`` doesn't sound right to me. I'd be fine changing the name in E.R2 later on; the main thing to me is that we have a function that does that thing.